tests/ord-service/tests/subscription_flow_test.go

```diff
@@ -276,15 +308,37 @@ func TestConsumerProviderFlow(stdT *testing.T) {
 		// Make a request to the ORD service with http client containing certificate with provider information and token with the consumer data.
 		stdT.Log("Getting consumer application using both provider and consumer credentials...")
 		respBody := makeRequestWithHeaders(stdT, certHttpClient, conf.ORDExternalCertSecuredServiceURL+"/systemInstances?$format=json", headers)
-		require.Equal(stdT, 1, len(gjson.Get(respBody, "value").Array()))
+		require.Len(stdT, gjson.Get(respBody, "value").Array(), 1)
 		require.Equal(stdT, consumerApp.Name, gjson.Get(respBody, "value.0.title").String())
 		stdT.Log("Successfully fetched consumer application using both provider and consumer credentials")
 
+		// Make a request to the ORD service expanding bundles and destinations.
+		// With no destinations
+		respBody = makeRequestWithHeaders(stdT, certHttpClient, conf.ORDExternalCertSecuredServiceURL+
+			"/systemInstances?$expand=consumptionBundles($expand=destinations)&$format=json", headers)
+
+		require.Len(stdT, gjson.Get(respBody, "value").Array(), 1)
+		require.Equal(stdT, consumerApp.Name, gjson.Get(respBody, "value.0.title").String())
+		require.NotEmpty(stdT, gjson.Get(respBody, "value.0.consumptionBundles.0.destinations").Raw)
+		require.Empty(stdT, gjson.Get(respBody, "value.0.consumptionBundles.0.destinations").Array())
+		stdT.Log("Successfully fetched system with bundles with no destinations")
+
+		// With destinations
+		respBody = makeRequestWithHeaders(stdT, certHttpClient, conf.ORDExternalCertSecuredServiceURL+
+			"/systemInstances?$expand=consumptionBundles($expand=destinations)&$format=json&reload=true", headers)
+		require.Equal(stdT, 1, len(gjson.Get(respBody, "value").Array()))
+		require.Equal(stdT, consumerApp.Name, gjson.Get(respBody, "value.0.title").String())
+		require.NotEmpty(stdT, gjson.Get(respBody, "value.0.consumptionBundles.0.destinations").Raw)
+		destinationsFromResponse := gjson.Get(respBody, "value.0.consumptionBundles.0.destinations").Array()
+		require.Len(stdT, destinationsFromResponse, 1)
+		require.Equal(stdT, destination.Name, destinationsFromResponse[0].Get("sensitiveData.destinationConfiguration.Name").String())
+		stdT.Log("Successfully fetched system with bundles and destinations")
```

Comment:
I assume that's the validation when subscribe request is executed - to create a destination? What's the case when someone has triggered an unsubscribe request - the destination should be deleted? If there is a logic on unsubscribe requests we could add additional checks/validation for the destinations that cover the unsubscribe flow as well? 

On line 337 we explicitly execute an unsubscribe request and below that is the validation regarding it.